File: 01 clase Entero/js/entero.js
class Enteros {
    // ATRIBUTO
    Num;
    //constructor = inicializa el objeto
    constructor(Numero) {
        this.Num=Numero
    }
    //class-cargar el valor de num
    SetNum(){
        const input =document.getElementById('numeroInput');
        this.Num=parseInt(input.value,10);
    }
    //class -obtener el valor de num
    getNum(){
      return this.Num;
    }
    //class -mostrar el vcalor de num
    showNum(){
        const resultadoDiv=document.getElementById("resultado");
        resultadoDiv.textContent=this.getNum();
    
    }
    showResultado(respuesta){
        const resultadoDiv=document.getElementById("resultado");
        resultadoDiv.textContent=respuesta;
    
    }
    incrementarNum(){
        this.Num =this.Num+1;
    }
    decrementarNum(){
        this.Num =this.Num-1;
    }
    esParImpar(){
        if(this.Num % 2 ==0){
            return true;

        }else{
            return false;
        }
    }
    esPositivoNegativo(){
        if(this.Num>0){
            return true;

        }else{
            return false;
        }

    }
    factorial() {
        if (this.Num < 0) {
            return "El factorial no está definido para números negativos.";
        }
        let resultado = 1;
        for (let i = 1; i <= this.Num; i++) {
            resultado *= i;
        }
        return resultado;
    }
    esPerfecto() {
        let sumaDivisores = 0;
        for (let i = 1; i < this.Num; i++) {
            if (this.Num % i === 0) {
                sumaDivisores += i;
            }
        }
        return sumaDivisores === this.Num;
    }
}

        


//las funciones =button HTML
var ClaseEnteros=new Enteros(0);
//crear el objeto

function cargarNum(){
    ClaseEnteros.SetNum();

}
function mostrarNum(){
    ClaseEnteros.showNum();

}
function incrementarvalor(){
    ClaseEnteros.incrementarNum();
    ClaseEnteros.showNum();

}
function decrementarvalor(){
    ClaseEnteros.decrementarNum();
    ClaseEnteros.showNum();

}
function esParImparNum(){
    var respuesta = ClaseEnteros.esParImpar();
    var resp =respuesta ? "es num par":"es num impar"
    ClaseEnteros.showResultado(resp)
}
function esPositivoNegativoNum(){
    var respuesta = ClaseEnteros.esPositivoNegativo();
    var resp =respuesta ? "es num positivo":"es num negativo"
    ClaseEnteros.showResultado(resp)
}
function FactorialNum(){
    var resultado = ClaseEnteros.factorial();
    ClaseEnteros.showResultado(resultado);
}
function esPerfectoNum () {
    var resultado = ClaseEnteros.esPerfecto();
    var resp = resultado ? "Es un número perfecto" : "No es un número perfecto";
    ClaseEnteros.showResultado(resp);
}

File: 01 clase Entero/entero.py
class Entero:
    def __init__(self,Numero):
        self.Num = Numero

    def setNum(self):
        input_value =input("enter a number")
        self.Num =int(input_value)
    
    def getNum(self):
        return self.Num
    
    def showNum (self):
        print(self.getNum())

    def showResultado(self,respuesta):
        print(respuesta)

    def incrementarNum (self):
        self.Num += 1

    def decrementarNum (self):
        self.Num -= 1 
    

    def esParImpar (self):
        return(self.Num% 2 == 0)
    
    def esPerfecto(self) :
        return sum( i for i in range (1, self.Num)if self.Num % i == 0) == self.Num
        
    
     
    
        
    
    def menu (self):
        while True:
            print("\nMenu:")
            print("1. establecer un nuevo numero")
            print("2.mostrar el numero actual")
            print("3.incrementar el numero")
            print("4.decrementar el numero")
            print("5. verificar si es par o impar")
            print("6.es perfecto")
            print("7.salir ")

            opcion=input("ingrese una opcion")
            if opcion == '1' :
                self.setNum()
            elif opcion == '2' :
                self.showNumNum()
            elif opcion == '3' :
                self.incrementarNum()
            elif opcion == '4' :
                self.decrementarNum()
            elif opcion == '5' :
                if self.esParImpar():
                    print("el numero es par ")
                else :
                    print("el numero es impar ")
            elif opcion == '6' :
                if self.esPerfecto():
                    print("el numero es perfecto ")
                else :
                    print("el numero no es  ")        
            elif opcion == '7':
                print("hasta luego")
                break
            else:
                print("opcion invalida,intente nuevamente")

numero=Entero(10)
numero.menu()
